### PR TITLE
feat(notifications): stabilize catchup logic and 48-hour planning chain

### PR DESCRIPTION
**Pull request by sanderw-be**
## Overview
This PR stabilizes the smart and catchup reminder notification architecture by finalizing the integration of catchup reminders into the 48-hour planning chain and unifying the triggering system to prevent duplicate notifications.

## Key Changes
- **48-Hour Planning:** Catchup reminders are now planned for both Today and Tomorrow, ensuring a continuous notification chain in the Native Alarm Bridge.
- **Stable Today Policy:** Current day's reminders are now \"locked\" and carried forward during re-plans, ensuring the user's schedule doesn't change unexpectedly.
- **Catchup Formula Fix:** Corrected the catchup evaluation to use `progressRatio <= expectedRatio`. This ensures catchups fire if the user is behind or exactly on track relative to reminders sent, but stops strictly when the daily goal is reached.
- **Duplicate Prevention:** Removed redundant Expo notification scheduling in `updateUpcomingReminderContent`. All smart/catchup triggers are now handled exclusively by the Native Alarm Bridge.
- **Defensive Headless Task:** Added robust error handling and database re-initialization guards in the background task to prevent crashes during shared object releases.

## Verification
- **Unit & Integration Tests:** 96 tests passing across `notificationManager.test.ts` and `smartReminderTask.test.ts`.
- **Edge Cases:** Verified `0/0` (send) and `30/30` (skip) catchup scenarios.

## Impact
- Eliminates duplicate notifications.
- Improves reliability of reminders when the app is in the background.
- Provides a more predictable and stable user experience for daily goals.

### CODE REVIEW

File: src/background/smartReminderTask.ts [hunk 1]
```diff
@@ -21,13 +21,36 @@ export const handleSmartReminder = async (data: HeadlessData) => {
 
   try {
     // 0. Initialize Infrastructure (Database + DI Container)
-    await initDatabaseAsync();
-    createContainer(db);
-
     const storageService = new StorageService(db);
```
[the remark this answers is not in the document]
Comment (sanderw-be) [reply src/background/smartReminderTask.ts]: Great catch. I've moved the `StorageService` instantiation after `initDatabaseAsync()` to ensure it receives the properly initialized `db` instance. Verified that background tasks now correctly access the database.

File: src/notifications/services/SmartReminderScheduler.ts [hunk 2]
```diff
@@ -322,20 +336,21 @@ export class SmartReminderScheduler implements ISmartReminderScheduler {
...
+          todayMinutes,
+          dailyTarget,
+          allPlannedItems,
+          newQueueEntries,
+          uiSlots
+        );
+
+        // If NO slots were carried forward (e.g. first time planning today),
+        // then we perform a full scoring plan.
+        if (newQueueEntries.length === 0) {
```
[the remark this answers is not in the document]
Comment (sanderw-be) [reply src/notifications/services/SmartReminderScheduler.ts]: Agreed. I've refactored the planning logic to support gap-filling for both smart and catchup reminders. The system now calculates the remaining capacity based on carried-forward slots and plans only the missing entries for the current day. This ensures a stable experience even when user settings change mid-day.

File: src/notifications/services/SmartReminderScheduler.ts [hunk 3]
```diff
@@ -552,7 +574,7 @@ export class SmartReminderScheduler implements ISmartReminderScheduler {
         let catchupAdded = 0;
         for (const slot of scores) {
           if (catchupAdded >= catchupLimit) break;
-          if (slot.score < 0.25) continue;
+          if (slot.score < 0.2) continue; // Lower threshold for catchup slots
```
[the remark this answers is not in the document]
Comment (sanderw-be) [reply src/notifications/services/SmartReminderScheduler.ts]: Done. I've introduced `MIN_SMART_SCORE_THRESHOLD` and `MIN_CATCHUP_SCORE_THRESHOLD` constants at the top of `SmartReminderScheduler.ts` to replace these magic numbers. verified and applied across both smart and catchup planning loops.
